src/App.tsx

```diff
@@ -1,12 +1,62 @@
 import React from 'react';
+import { GoodsList } from './components/GoodsList';
 import './App.scss';
 
-// import { getAll, get5First, getRed } from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
 // or
 // import * as goodsAPI from './api/goods';
 
-const App: React.FC = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+interface Good {
+  id: number,
+  name: string,
+  color: string,
+}
+
+interface State {
+  goods: Good[],
+}
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    goods: [],
+  };
+
+  handleClick = async (callback: () => Promise<Good[]>) => {
```

Comment:
Main purpose of this function is to get goods from server.
```suggestion
  getGoods = async (callback: () => Promise<Good[]>) => {
```